### PR TITLE
[readtapi] Use OptParser.td for options

### PR DESCRIPTION
**Pull request by cyndyishida**
This includes revamping how --compare was handled and adds `-o` ontop of tablegen approach.

This will be used to add more complex options.

### CODE REVIEW

File: llvm/test/tools/llvm-readtapi/command-line.test [hunk 1]
```diff
@@ -0,0 +1,7 @@
+; RUN: llvm-readtapi --help 2>&1 | FileCheck %s 
+; RUN: llvm-readtapi -help 2>&1 | FileCheck %s 
+
+CHECK: OVERVIEW: LLVM TAPI file reader and manipulator
+CHECK: USAGE: llvm-readtapi [options] <inputs>
+CHECK: OPTIONS:
+CHECK:   -help    display this help
```
Comment (ributzka) [inline llvm/test/tools/llvm-readtapi/command-line.test:7]: Does it also emit help for the compare option?

Comment (cyndyishida) [reply llvm/test/tools/llvm-readtapi/command-line.test:7]: Yes, I didn't add the others so I wouldn't need to update it as more options get added.

File: llvm/tools/llvm-readtapi/Opts.td
Comment (ributzka) [inline llvm/tools/llvm-readtapi/Opts.td:1]: `Opts.td` is a very generic file name. Could this be renamed to for example `TapiOpts.td`?

File: llvm/tools/llvm-readtapi/llvm-readtapi.cpp [hunk 2]
```diff
@@ -11,80 +11,132 @@
...
-  cl::HideUnrelatedOptions(TapiCat);
-  cl::ParseCommandLineOptions(Argc, Argv,
-                              "TAPI File Reader and Manipulator Tool");
-
-  if (Output == OutputKind::Compare) {
-    if (InputFileName.empty() || CompareInputFileName.empty()) {
-      cl::PrintHelpMessage();
-      return EXIT_FAILURE;
-    }
+const int NON_TAPI_EXIT_CODE = 2;
```
Comment (ributzka) [inline llvm/tools/llvm-readtapi/llvm-readtapi.cpp:73]: What is `NON_TAPI_EXIT_CODE`?

File: llvm/tools/llvm-readtapi/Opts.td
Comment (cyndyishida) [reply llvm/tools/llvm-readtapi/Opts.td:1]: Yes, I was mostly following a pattern set by other tools. But it's not super consistent anyway. 

```
/Projects/llvm/llvm-project/llvm/tools
$ fd Opts.td
llvm-cvtres/Opts.td
llvm-cxxfilt/Opts.td
llvm-debuginfod/Opts.td
llvm-dwp/Opts.td
llvm-gsymutil/Opts.td
llvm-ifs/Opts.td
llvm-libtool-darwin/Opts.td
llvm-lipo/LipoOpts.td
llvm-ml/Opts.td
llvm-mt/Opts.td
llvm-nm/Opts.td
llvm-objcopy/BitcodeStripOpts.td
llvm-objcopy/CommonOpts.td
llvm-objcopy/InstallNameToolOpts.td
llvm-objcopy/ObjcopyOpts.td
llvm-objcopy/StripOpts.td
llvm-objdump/ObjdumpOpts.td
llvm-objdump/OtoolOpts.td
llvm-rc/Opts.td
llvm-rc/WindresOpts.td
llvm-readobj/Opts.td
llvm-readtapi/Opts.td
llvm-size/Opts.td
llvm-strings/Opts.td
llvm-symbolizer/Opts.td
llvm-tli-checker/Opts.td
sancov/Opts.td
```

File: llvm/tools/llvm-readtapi/llvm-readtapi.cpp (hunk 2 above)
Comment (cyndyishida) [reply llvm/tools/llvm-readtapi/llvm-readtapi.cpp:73]: `readtapi --compare` behaves like `diff` where exit codes differentiate between the same/different/other kind of error (e.g. missing file) 
I can add a comment to make that clearer.

**Review by ributzka — APPROVED** [review]
LGTM. Thanks Cyndy

**Pull request merged**